xknx/telegram/telegram.py

```diff
@@ -52,15 +52,17 @@ def __init__(
         self.timestamp = datetime.now()
         self.tpci: TPCI
         if tpci is None:
-            if isinstance(destination_address, GroupAddress):
-                if destination_address.raw == 0:
-                    self.tpci = TDataBroadcast()
-                else:
-                    self.tpci = TDataGroup()
-            elif isinstance(destination_address, IndividualAddress):
-                self.tpci = TDataIndividual()
-            else:  # InternalGroupAddress
```

Comment:
🧑‍💻
This is sorted by the most common use case - so first GroupAddress is handled. Lastly InternalGroupAddress as this is hardly used anyway and more of an edge-case.
Other than that we could firstly check for `IndividualAddress`, then for `destination_address.raw == 0` and assume all other are `TDataGroup()`.